libtransmission/rpcimpl.c

```diff
@@ -944,26 +944,32 @@ static char const* torrentGet(tr_session* session, tr_variant* args_in, tr_varia
 
 static char const* setLabels(tr_torrent* tor, tr_variant* list)
 {
-    char const* str;
-    size_t str_len;
     int const n = tr_variantListSize(list);
     char const* errmsg = NULL;
     tr_ptrArray labels = TR_PTR_ARRAY_INIT;
     int labelcount = 0;
     for (int i = 0; i < n; i++)
     {
+        char const* str;
+        size_t str_len;
         if (tr_variantGetStr(tr_variantListChild(list, i), &str, &str_len) && str != NULL && str_len != 0)
         {
+            char* label = tr_strndup(str, str_len);
+            tr_strstrip(label);
+            if (*label == '\0')
+            {
+                errmsg = "labels cannot be empty";
+            }
+
             if (strchr(str, ',') != NULL)
```

Comment:
Minor, but I'd stop on the first error we get, i.e.
```suggestion
             if (errmsg != NULL && strchr(str, ',') != NULL)
```

Same below: could wrap following `for` and `if` in `if (errmsg != NULL)` as well.

Reply:
Looks like I'm an idiot once again. It should've been `errmsg == NULL` :)